src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/AnalysisEntity.cs

```diff
@@ -204,6 +204,30 @@ or PointsToAbstractValueKind.UnknownNull
 
         public bool IsLValueFlowCaptureEntity => CaptureId.HasValue && CaptureId.Value.IsLValueFlowCapture;
 
+        public bool EqualsIgnoringIndices(AnalysisEntity? other)
+        {
+            // Perform fast equality checks first.
+            if (ReferenceEquals(this, other))
+            {
+                return true;
+            }
+
+            if (other == null ||
+                EqualsIgnoringInstanceLocationId != other.EqualsIgnoringInstanceLocationId)
+            {
+                return false;
+            }
+
+            // Now perform slow check that compares individual hash code parts sequences.
```

Comment:
Why are we comparing hash codes?

Reply:
@Youssef1313 We were seeing extremely large allocations and CPU stacks for dictionary operations keyed on `AnalysisEntity` during flow analysis, hence we had switched to using a CacheBasedEquatable approach. It's possible that this is no longer required in the current state of the flow analysis framework, but we would need to invest a few cycles in cleanup and performance measurements on real world code to confirm.